private/buf/cmd/buf/command/push/push.go

```diff
@@ -168,25 +169,67 @@ func run(
 	if err != nil {
 		return err
 	}
-	protoModule, err := bufmodule.ModuleToProtoModule(ctx, builtModule.Module)
+	modulePin, err := push(ctx, container, moduleIdentity, builtModule, flags)
 	if err != nil {
+		if connect.CodeOf(err) == connect.CodeAlreadyExists {
+			if _, err := container.Stderr().Write(
+				[]byte("The latest commit has the same content; not creating a new commit.\n"),
+			); err != nil {
+				return err
+			}
+			return nil
+		}
 		return err
 	}
-	var (
-		bucketManifest *modulev1alpha1.Blob
-		blobs          []*modulev1alpha1.Blob
-	)
-	if enabled, err := strconv.ParseBool(container.Env(bufcli.BetaEnableTamperProofingEnvKey)); err == nil && enabled {
-		bucketManifest, blobs, err = manifestAndFilesBlobs(ctx, builtModule.Bucket)
-		if err != nil {
-			return err
-		}
+	if modulePin == nil {
+		return errors.New("Missing local module pin in the registry's response.")
+	}
+	if _, err := container.Stdout().Write([]byte(modulePin.Commit + "\n")); err != nil {
+		return err
 	}
+	return nil
+}
+
+func push(
+	ctx context.Context,
+	container appflag.Container,
+	moduleIdentity bufmoduleref.ModuleIdentity,
+	builtModule *bufmodulebuild.BuiltModule,
+	flags *flags,
+) (*registryv1alpha1.LocalModulePin, error) {
 	clientConfig, err := bufcli.NewConnectClientConfig(container)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	service := connectclient.Make(clientConfig, moduleIdentity.Remote(), registryv1alpha1connect.NewPushServiceClient)
+	// Check if tamper proofing env var is enabled
+	tamperProofingEnabled, err := strconv.ParseBool(container.Env(bufcli.BetaEnableTamperProofingEnvKey))
+	if err == nil && tamperProofingEnabled {
```

Comment:
Nits:

### `if err != nil`
- should we log it?, so a user with that ENV flag set to true would know that a regular push was used instead?
- should we fail and don't even try the push? otherwise how a user with that ENV set knows for sure the new RPC was used? how brittle is `strconv.ParseBool(container.Env(bufcli.BetaEnableTamperProofingEnvKey))`?

### `if tamperProofingEnabled`
- Log that the new RPC is being used? so the user with that ENV set knows for sure that's being used. This log could be removed whenever we turn on this by default.

Reply:
Good point on the `if err != nil` - updated to better report errors to the user so we don't switch behavior silently if the env var contains an invalid value. On the `if ...` logging, I'd prefer we wait on this (I expect us to provide this to a smaller number of users and then probably remove the flag once we feel ok).